cloudant/CDTDatastore
# 45570 Generate, retrieve & store a key: Manager to generate & retrieve a key

*What:*
Define a manager able to create a DPK (Data Protection Key) and store and retrieve this DPK from keychain.

*Why:*
This PR is the last one in a bigger development to provide a mechanism to generate a key and safely store it.

*How:*
The manager will be based on the classes implemented before to apply AES, PBKDF2 and read from/write to the keychain.

*Tests:*
* Clear the data stored in the keychain
* Check there is data in the keychain
* Get data from the keychain
* Check that the number of iterations stored in the keychain is right
* Check that the IV in the keychain has the correct size
* Validate the steps followed to retrieve a key
* Check that is not possible to generate the key twice
* Validate the steps followed to generate a key

reviewer @mikerhodes 
reviewer @rhyshort 

**NOTE:** Wait to review this PR until PR #134  is closed. This branch will be rebased and there will be only one commit to review.
